### ISSUE TITLE
Generated rbac rules lack API group domain

### ISSUE THREAD

**Issue by sttts** [kubernetes-sigs/kubebuilder#338]
I have generated a project with:

```shell
$ kubebuilder init --domain k8s-go.openshift.org --license apache2 --owner "The workshop members"
$ kubebuilder create api --group policy --version v1beta1 --kind HealthCheckPolicy
```

The generated rbac role is wrong (lacks the `k8s-go.openshift.org` domain):

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: null
  name: manager-role
rules:
- apiGroups:
  - apps
  resources:
  - deployments
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
- apiGroups:
  - policy
  resources:
  - healthcheckpolicies
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
```

**Comment by droot** [comment]
I am able to reproduce it. 
The reason is that the scaffolded controller `pkg/controller/healthcheckpolicy/healthcheckpolicy_controller.go` does not have `groups` part correct in the RBAC annotation. Pasting the content below. So instead of `groups=policy`, it should be `groups=policy.k8s-go.openshift.org`.

```go
// +kubebuilder:rbac:groups=policy,resources=healthcheckpolicies,verbs=get;list;watch;create;update;patch;delete
func (r *ReconcileHealthCheckPolicy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
        // Fetch the HealthCheckPolicy instance
        instance := &policyv1beta1.HealthCheckPolicy{}
```
@fanzhangio would you like to take a look at the scaffolding code for the RBAC annotation since you are more familiar with it ?

**Comment by fanzhangio** [comment]
Yes, I have identified the cause of bug, will fix it in a PR.

**Comment by droot** [comment]
This should be fixed with #343
